devito/ir/clusters/cluster.py

```diff
@@ -149,6 +149,10 @@ def functions(self):
     def has_increments(self):
         return any(e.is_Increment for e in self.exprs)
 
+    @cached_property
```

Comment:
Again, shouldn't implicit dims be handled prior to this point to avoid needing this property?